x-pack/plugins/file_upload/public/util/file_parser.js

```diff
@@ -7,111 +7,136 @@
+    for await (const batch of batches) {
+      if (getFileParseActive()) {
+        switch (batch.batchType) {
+          case 'root-object-batch-complete':
+            if (getFileParseActive()) {
+              if (featuresProcessed) {
+                parsedGeojson = { ...batch.container, features };
+              } else {
+                // Handle single feature geoJson
+                try {
+                  parsedGeojson = cleanAndValidate(batch.container);
+                  if (parsedGeojson) {
+                    featuresProcessed++;
+                  }
+                } catch (e) {
+                  boolCleaningErrs = true;
+                  errors.push(e);
+                  resolve(null);
+                  return;
+                }
+              }
+            } else {
+              resolve(null);
+              return;
+            }
+            break;
+          default:
+            for (const feature of batch.data) {
+              let cleanFeature;
+              try {
+                cleanFeature = cleanAndValidate(feature);
+                if (cleanFeature) {
+                  features.push(cleanFeature);
+                }
+              } catch (e) {
+                // TODO: Give feedback on which features failed
+                if (!boolCleaningErrs) {
+                  boolCleaningErrs = true;
+                  errors.push(
+                    new Error(
+                      i18n.translate('xpack.fileUpload.fileParser.cleaningErrors', {
+                        defaultMessage: 'Some features with cleaning errors omitted',
+                      })
+                    )
+                  );
+                }
+              }
+              // TODO: Give feedback on which features failed
+              if ((!feature.geometry || !feature.geometry.type) && !boolGeometryErrs) {
+                boolGeometryErrs = true;
+                errors.push(
+                  new Error(
+                    i18n.translate('xpack.fileUpload.fileParser.featuresOmitted', {
+                      defaultMessage: 'Some features without geometry omitted',
+                    })
+                  )
+                );
+              }
+              featuresProcessed++;
+            }
         }
         setFileProgress({
           featuresProcessed,
-          bytesProcessed: stop || file.size,
+          bytesProcessed: batch.bytesUsed,
           totalBytes: file.size,
         });
-        patternReader.writeDataToPatternStream(result);
-        if (!stop) {
-          return;
-        }
-
-        start = stop;
-        const newStop = stop + FILE_BUFFER;
-        // Check EOF
-        stop = newStop > file.size ? undefined : newStop;
-        readSlice(fileReader, file, start, stop);
+      } else {
+        break;
       }
-    };
-    fileReader.onerror = () => {
-      fileReader.abort();
-      patternReader.abortStream();
+    }
+
+    if (!featuresProcessed && getFileParseActive()) {
       reject(
```

Comment:
I would expect to reach this error if invalid json or a non GeoJSON file is provided.